packages/material-ui-system/src/spacing.js

```diff
@@ -0,0 +1,140 @@
+import warning from 'warning';
+import resonsivePropType from './resonsivePropType';
+import { handleBreakpoints } from './breakpoints';
+import merge from './merge';
+import memoize from './memoize';
+
+const properties = {
+  m: 'margin',
+  p: 'padding',
+};
+
+const directions = {
+  t: 'Top',
+  r: 'Right',
+  b: 'Bottom',
+  l: 'Left',
+  x: ['Left', 'Right'],
+  y: ['Top', 'Bottom'],
+};
+
+// memoize() impact:
+// From 300,000 ops/sec
+// To 350,000 ops/sec
+const getCssProperties = memoize(prop => {
+  const [a, b] = prop.split('');
+  const property = properties[a];
+  const direction = directions[b] || '';
+  return Array.isArray(direction) ? direction.map(dir => property + dir) : [property + direction];
+});
+
+const spacingKeys = [
+  'm',
+  'mt',
+  'mr',
+  'mb',
+  'ml',
+  'mx',
+  'my',
+  'p',
+  'pt',
+  'pr',
+  'pb',
+  'pl',
+  'px',
+  'py',
+];
+
+function getTransformer(theme) {
+  const themeTransformer =
+    theme.spacing && theme.spacing.unit ? theme.spacing.unit : theme.spacing || 8;
```

Comment:
```suggestion
    theme.spacing && theme.spacing.unit != null ? theme.spacing.unit : theme.spacing != null ? theme.spacing : 8;
```
Either this or use the suggested warning message.